### PR TITLE
(HC-57) Add render for config_value family

### PR DESCRIPTION
**Pull request by HAIL9000**
Implement the render method for the family of classes that inherit
from the config_value class.

### CODE REVIEW

File: lib/src/values/simple_config_list.cc [hunk 1]
```diff
@@ -99,7 +99,9 @@ namespace hocon {
             }
 
             if (equal(begin(), end(), o.begin(),
-                      [](shared_value const& a, shared_value const& b) { return a == b; })) {
+                      [](shared_value const& a, shared_value const& b) {
```
Comment (MikaelSmith) [inline lib/src/values/simple_config_list.cc]: I think you misunderstood the function here. It checks if the pointers are equal, then it does a deep comparison below if that failed.

File: lib/tests/path_test.cc [hunk 2]
```diff
@@ -31,6 +31,7 @@ TEST_CASE("path to string", "[path]") {
 
 void render_test(path p, string expected) {
     REQUIRE(p.render() == expected);
+    auto parsed = path_parser::parse_path(expected);
```
Comment (MikaelSmith) [inline lib/tests/path_test.cc]: I don't think you meant to leave this in.

File: lib/src/values/simple_config_object.cc [hunk 3]
```diff
@@ -390,4 +400,104 @@ namespace hocon {
         }
     }
 
+    bool compare(string& a, string& b) {
```
Comment (MikaelSmith) [inline lib/src/values/simple_config_object.cc]: http://en.cppreference.com/w/cpp/algorithm/sort notes that this function should use `string const&` arguments. Specifically, the signature is `bool cmp(const Type1 &a, const Type2 &b);`. That's probably what's causing the compile error.

**Comment by HAIL9000** [comment]
@MikaelSmith Comments addressed!

**Comment by MikaelSmith** [comment]
👍

File: lib/tests/conf_parser_test.cc [hunk 4]
```diff
@@ -47,13 +47,13 @@ TEST_CASE("valid conf works") {
...
         // let's also check round-trip rendering
-        //auto rendered = our_ast->render();
-        //CAPTURE(rendered);
-        //auto reparsed = parse(rendered);
-        //REQUIRE(*our_ast == *reparsed);
+        auto rendered = our_ast->render();
+        CAPTURE(rendered);
+        auto reparsed = parse(rendered);
+        auto foo = *our_ast == *reparsed;
+        REQUIRE(foo);
```
Comment (branan) [inline lib/tests/conf_parser_test.cc]: "foo"?

**Comment by branan** [comment]
Other than squashing that REQUIRE down to one line (or giving it a better variable name) I'm :+1:

**Pull request merged**